pennylane_lightning/core/src/observables/Observables.hpp

```diff
@@ -140,6 +155,33 @@ class NamedObsBase : public Observable<StateVectorT> {
     void applyInPlace(StateVectorT &sv) const override {
         sv.applyOperation(obs_name_, wires_, false, params_);
     }
+
+    void
+    applyInPlace(StateVectorT &sv, bool shots,
+                 std::vector<size_t> &identify_wire,
+                 std::vector<size_t> &ob_wires = {},
+                 [[maybe_unused]] const size_t term_idx = 0) const override {
+        ob_wires = getWires();
+        if (shots) {
+            if (obs_name_ == "PauliX") {
+                sv.applyOperation("Hadamard", wires_, false);
+            } else if (obs_name_ == "PauliY") {
+                sv.applyOperations({"PauliZ", "S", "Hadamard"},
+                                   {wires_, wires_, wires_},
+                                   {false, false, false});
+            } else if (obs_name_ == "Hadamard") {
+                const PrecisionT theta = -M_PI / 4.0;
+                sv.applyOperation("RY", wires_, false, {theta});
+            } else if (obs_name_ == "PauliZ") {
+            } else if (obs_name_ == "Identity") {
+                if (!identify_wire.empty()) {
+                    identify_wire.clear();
+                }
+                identify_wire.push_back(wires_[0]);
+            } else {
```

Comment:
You may want to add `// TODO` for the future reference.

Reply:
Let's add a PL_ABORT here and the error message will let users know what are the obs supported.